### PR TITLE
Docs/update local setup doc

### PR DESCRIPTION
**Pull request by cmccormack**
`npm ci` and `git clone --depth=1` commands were recommended by @raisedadead during conversation in the [gitter Contributor's channel](https://gitter.im/FreeCodeCamp/Contributors?at=5c8ab5c3b4b6ef7bc8ad610e).

The main goal of this PR is to clean up as much grammar/formatting as possible, improve readability, and add a few commands that should improve the experience and prevent potential issues in the future.  I also replaced many bold section "headers" with actual headers so it's easier to share just that child section vs the entire parent section.

- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [X] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

**Comment by raisedadead** [comment]
Has a PR that will need some update https://github.com/freeCodeCamp/freeCodeCamp/pull/35162

**Comment by raisedadead** [comment]
@cmccormack can you update the docs to reflect the latest changes? Thanks a lot.

File: docs/how-to-setup-freecodecamp-locally.md
Comment (raisedadead) [inline docs/how-to-setup-freecodecamp-locally.md]: One concern that I have with this, subsequent installs.

That is if someone does add a new package or updates a package as a part of the a legitimate change, `npm ci` will blow away the `node_modules`.

This would mean instead of an update it would become more slower to update the local setup. But that just circles around to the original problem.

Comment (cmccormack) [reply docs/how-to-setup-freecodecamp-locally.md]: It blows it away but it is very fast.  I would recommend we do some testing though if there are concerns

Comment (raisedadead) [reply docs/how-to-setup-freecodecamp-locally.md]: Yup, I did some research. And it's going to be fine. Since all it does is look at the lockfile which in turn has a version of a package pinned, it means that almost every time the package is going to be there in the cache that npm has already created on your local machine.

Let's proceed. I am quite happy with this change.

**Review by Nirajn2311 — APPROVED** [review]
LGTM 👍 
I tried `npm ci` and it was fast, almost cutting down the time to half as before 😃 .

**Comment by Nirajn2311** [comment]
@raisedadead @tchaffee Waiting for your approval.

**Comment by tchaffee** [comment]
This looks good to me. Is there a plan to make all these changes in `/docs/how-to-setup-freecodecamp-locally-using-docker.md` too?

**Review by raisedadead — APPROVED** [review]

**Pull request merged**
